Fix slider repeats playing sample twice

Shawdooow opened the pull request:
they are no longer glued to the end circle only + sound fix (their sound was being played twice and was noisey)

Shawdooow committed:
Merge remote-tracking branch 'refs/remotes/ppy/master'

Shawdooow committed:
the fix

Shawdooow committed:
fix the fix

silly me, that was configured for vitaru!

Shawdooow committed:
fix the fix harder

Shawdooow committed:
dont play the same sound twice at once

kmoore43 commented on osu.Game.Rulesets.Osu/Objects/Slider.cs:
```diff
@@ -155,13 +155,20 @@ private void createRepeatPoints()
             var repeatPointDistance = Math.Min(Distance, length);
             var repeatDuration = length / Velocity;
 
+            bool sliderStart = true;
```
Since this is local, I don't see it being set anywhere else. If that's the case is it required?

peppy commented on osu.Game.Rulesets.Osu/Objects/Slider.cs:
```diff
@@ -155,13 +155,20 @@ private void createRepeatPoints()
                     var distanceProgress = d / length;
 
+                    if (sliderStart)
+                        distanceProgress = 0;
```
why is this needed?

Shawdooow replied:
this makes even numbered repeats where the start circle is like normal sliders, before they were always at the end circle

smoogipoo committed:
Merge remote-tracking branch 'origin/master' into sliderbouncers-fix

smoogipoo committed:
Update framework

smoogipoo committed:
Add a method to get ruleset by name

smoogipoo committed:
Add HitCircle/Slider testcases

smoogipoo committed:
Finish up implementation of TestCaseSlider

smoogipoo committed:
Cleanups + add auto to TestCaseHitCircle

smoogipoo committed:
Add a Spinner testcase

smoogipoo committed:
Move sliderbars to the end of the testcase buttons

smoogipoo committed:
Decomission TestCaseHitObjects

smoogipoo committed:
Simplify creation of repeat points

smoogipoo committed:
Fix repeat points always being placed at end position of th slider

PositionAt is clamped to [0, 1] where 0 denotes the beginning of the curve and 1 denotes the end of the curve. It has no concept of repeats.

smoogipoo committed:
Ignore test cases for CI

smoogipoo commented:
@Shawdooow oops I kind of abused this branch, sry. @peppy I don't think I'm eligible to approve this anymore :P

smoogipoo committed:
Display larger hitcircles

peppy committed:
Merge branch 'master' into sliderbouncers-fix
